tests/test_hub_features.py

```diff
@@ -35,3 +38,34 @@ def test_subfolder(self):
             model = PeftModel.from_pretrained(model, model_id, subfolder=subfolder)
 
             assert isinstance(model, PeftModel)
+
+
+class TestBaseModelRevision:
+    def test_save_and_load_base_model_revision(self, tmp_path):
+        r"""
+        Test if subfolder argument works as expected
+        """
+        lora_config = LoraConfig(r=8, lora_alpha=16, lora_dropout=0.0, init_lora_weights=False)
```

Comment:
I tried this test on the current main and it succeeds. It's thus not testing what it's supposed to test.

One issue is that without setting a seed, calling `get_peft_model` twice does not yield the same model, so that part of the test passes trivially.

I tried rewriting the test like this:

```python
        lora_config = LoraConfig(r=8, lora_alpha=16, lora_dropout=0.0, init_lora_weights=False)
        test_inputs = torch.arange(10).reshape(-1, 1)

        base_model_revision = AutoModelForCausalLM.from_pretrained("peft-internal-testing/tiny-random-BertModel", revision="v2.0.0").eval()
        torch.manual_seed(0)
        peft_model_revision = get_peft_model(base_model_revision, lora_config)
        output = peft_model_revision(test_inputs).logits

        # sanity check: the model without revision should be different
        base_model_no_revision = AutoModelForCausalLM.from_pretrained("peft-internal-testing/tiny-random-BertModel").eval()
        torch.manual_seed(0)
        peft_model_no_revision = get_peft_model(base_model_no_revision, lora_config)
        output_no_revision = peft_model_no_revision(test_inputs).logits
        assert not torch.allclose(output, output_no_revision)

        # check that if we save and load the model, the output corresponds to the one with revision
        peft_model_revision.save_pretrained(tmp_path / "base_model_revision")
        peft_model_loaded = AutoPeftModelForCausalLM.from_pretrained(tmp_path / "base_model_revision").eval()
        assert peft_model_loaded.peft_config["default"].revision == "v2.0.0"

        output_loaded = peft_model_loaded(test_inputs).logits
        assert torch.allclose(output, output_loaded)

```

Does this better capture what we want to test, WDYT? Note, however, that this test still fails for me even on your branch, because the stored `adapter_config.json` does not seem to contain the revision.

Reply:
Those are good points, thank you, I almost all the suggestions here.

I decided to remove `init_lora_weights=False` so it results in an identity transform and then we don't need to do `torch.manual_seed` as the peft model is the same as the base model, removing stochasticity